types/event-source-polyfill/index.d.ts

```diff
@@ -0,0 +1,82 @@
+// Type definitions for event-source-polyfill 1.0
+// Project: https://github.com/Yaffle/EventSource
+// Definitions by: Emily Klassen <https://github.com/forivall>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.4
+
+declare global {
+    // Declare empty stub interfaces for environments where "dom" lib is not included
+    interface EventSource {}
+    interface EventSourceInit {}
+}
+
+export interface EventSourcePolyfillInit {
+    withCredentials?: boolean;
+    lastEventIdQueryParameterName?: string;
+    heartbeatTimeout?: number;
+    headers?: { [name: string]: string };
+    Transport?: new () => any;
+}
+
+export interface Event {
+    type: string;
+    target: any;
+}
+
+export interface EventListener {
+    (evt: Event): void;
+}
+
+export interface EventListenerObject {
+    handleEvent(evt: Event): void;
+}
+
+export type EventListenerOrEventListenerObject = EventListener | EventListenerObject;
+
+export interface EventSourceEventMap {
+    "error": Event;
+    "message": MessageEvent;
+    "open": Event;
+}
+
+export interface MessageEvent extends Event {
+    data: any;
+    lastEventId: string;
+}
+
+export class EventSourcePolyfill {
+    static readonly CLOSED: number;
+    static readonly CONNECTING: number;
+    static readonly OPEN: number;
+
+    constructor(url: string, options: EventSourcePolyfillInit);
```

Comment:
```suggestion
    constructor(url: string, options?: EventSourcePolyfillInit);
```
are options required?

Reply:
my mistake. they're optional.